components/core/src/clp/DictionaryConcepts.hpp

```diff
@@ -0,0 +1,262 @@
+    } -> std::same_as<void>;
+
+    /**
+     * Adds an integer variable placeholder to the constant part of the logtype.
+     */
+    {
+        entry.add_int_var()
+    } -> std::same_as<void>;
+
+    /**
+     * Adds a float variable placeholder to the constant part of the logtype.
+     */
+    {
+        entry.add_float_var()
+    } -> std::same_as<void>;
+
+    /**
+     * Adds a dictionary variable placeholder to the constant part of the logtype.
+     */
+    {
+        entry.add_dictionary_var()
+    } -> std::same_as<void>;
+
+    /**
+     * @return The constant part of the logtype.
+     */
+    {
+        entry.get_value()
+    } -> std::same_as<std::string const&>;
+
+    /**
+     * @return The number of variables in the constant part of the logtype.
+     */
+    {
+        entry.get_num_variables()
+    } -> std::same_as<size_t>;
+
+    /**
+     * @return The number of variable placeholders (including escaped ones) in the constant part of
+     * the logtype.
+     */
+    {
+        entry.get_num_placeholders()
+    } -> std::same_as<size_t>;
+
+    /**
+     * Gets the position and type of a variable placeholder in the logtype.
+     * @param placeholder_ix The index of the placeholder to get the info for.
+     * @param placeholder_ref The type of the placeholder at `placeholder_ix`.
+     * @return The placeholder's position in the logtype, or `SIZE_MAX` if `placeholder_ix` is out
+     * of bounds.
+     */
+    {
+        entry.get_placeholder_info(placeholder_ix, placeholder_ref)
+    } -> std::same_as<size_t>;
+
+    /**
+     * @return The dictionary Id for this logtype.
+     */
+    {
+        entry.get_id()
+    } -> std::same_as<logtype_dictionary_id_t>;
+};
+
+/**
+ * Requirement for the variable dictionary entry interface.
+ * @tparam VariableDictionaryEntryType The type of the variable dictionary entry.
+ */
+template <typename VariableDictionaryEntryType>
+concept VariableDictionaryEntryReq = requires(VariableDictionaryEntryType entry) {
+    /**
+     * @return The dictionary Id for this variable.
+     */
+    {
+        entry.get_id()
+    } -> std::same_as<variable_dictionary_id_t>;
+};
+
+/**
+ * Requirement for the logtype dictionary reader interface.
+ * @tparam LogTypeDictionaryReaderType The type of the logtype dictionary reader.
+ * @tparam LogTypeDictionaryEntryType The type of the entries in the logtype dictionary reader.
+ */
+template <
+        typename LogTypeDictionaryReaderType,
+        typename LogTypeDictionaryEntryType = typename LogTypeDictionaryReaderType::entry_t>
+concept LogTypeDictionaryReaderReq = requires(
+        LogTypeDictionaryReaderType reader,
+        std::string_view logtype,
+        bool ignore_case,
+        std::unordered_set<LogTypeDictionaryEntryType const*>& entries
+) {
+    /**
+     * Gets entries matching a given logtype.
+     * @param logtype
+     * @param ignore_case Whether the lookup should be case insensitive.
```

Comment:
```suggestion
     * @param ignore_case Whether the search should be case insensitive.
```
Just to be consistent with the docstring below.